pkg/asset/installconfig/azure/client.go

```diff
@@ -32,9 +32,10 @@ type API interface {
 	ListResourceIDsByGroup(ctx context.Context, groupName string) ([]string, error)
 	GetStorageEndpointSuffix(ctx context.Context) (string, error)
 	GetDiskEncryptionSet(ctx context.Context, subscriptionID, groupName string, diskEncryptionSetName string) (*azenc.DiskEncryptionSet, error)
-	GetHyperVGenerationVersion(ctx context.Context, instanceType string, diskType string, region string) (string, error)
+	GetHyperVGenerationVersion(ctx context.Context, instanceType string, region string) (string, error)
```

Comment:
We do not necessarily need to keep the `GetHyperVGenerationVersion` as `GetVMCapabilities` is a more generalized solution that could replace it.

Reply:
I could not come up with a clean way of removing this `GetHyperVGenerationVersion` as it is called in a couple of different places. In machine, the variable is used to generate a name for the the image (https://github.com/openshift/installer/blob/master/pkg/asset/machines/azure/machines.go#L98-L99). Then later on it is used again to pass data into the terraform (https://github.com/openshift/installer/blob/master/pkg/asset/cluster/tfvars.go#L331-L332). 

Maybe if the terraform (tfvars.go) bits could look at the image value and determine what the proper value is for hypervgen?

Reply:
> I could not come up with a clean way of removing this `GetHyperVGenerationVersion` as it is called in a couple of different places. 

That's fine. It's probably best to try to keep this all as concentrated as possible.

> Maybe if the terraform (tfvars.go) bits could look at the image value and determine what the proper value is for hypervgen?

We went back and forth on this in the original PR and ultimately ended up with the current approach.

Reply:
Related to my previous comment, I think you can just drop this change from this PR now. Feel free to do it in a follow-up PR.